src/GraphControl/Control/Grapher.cpp

```diff
@@ -433,6 +439,105 @@ namespace GraphControl
         }
     }
 
+    shared_ptr<IGraph> Grapher::GetGraph(GraphControl::Equation ^ equation)
+    {
+        std::shared_ptr<Graphing::IGraph> graph = m_solver->CreateGrapher();
+
+        wstringstream ss{};
+        ss << s_getGraphOpeningTags;
+        ss << equation->GetRequest();
+        ss << s_getGraphClosingTags;
+
+        wstring request = ss.str();
+        unique_ptr<IExpression> graphExpression;
+        if (graphExpression = m_solver->ParseInput(request))
+        {
+            if (graph->TryInitialize(graphExpression.get()))
+            {
+                return graph;
+            }
+        }
+
+        return nullptr;
+    }
+
+    void Grapher::UpdateKeyGraphFeatures()
+    {
+        auto equations = GetValidEquations();
+        for (auto equation : GetValidEquations())
+        {
+            if (auto graph = GetGraph(equation))
+            {
+                if (auto analyzer = graph->GetAnalyzer())
+                {
+                    if (analyzer->CanFunctionAnalysisBePerformed())
+                    {
+                        m_solver->FormatOptions().SetFormatType(FormatType::MathML);
+                        if (S_OK
+                            == analyzer->PerformFunctionAnalysis(
+                                (Graphing::Analyzer::NativeAnalysisType)Graphing::Analyzer::PerformAnalysisType::PerformAnalysisType_All))
+                        {
+                            Graphing::IGraphFunctionAnalysisData functionAnalysisData = m_solver->Analyze(analyzer.get());
+                            {
+                                equation->XIntercept = ref new String(functionAnalysisData.Zeros.c_str());
+                                equation->YIntercept = ref new String(functionAnalysisData.YIntercept.c_str());
+                                equation->Domain = ref new String(functionAnalysisData.Domain.c_str());
+                                equation->Range = ref new String(functionAnalysisData.Range.c_str());
+                                equation->Parity = functionAnalysisData.Parity;
+
+                                // There is only ever one item in Periodicity. Map is being used because it works with DependencyProperty.
```

Comment:
Could we simplify this by just having a property for the periodicity and the direction?